vertical-pod-autoscaler/enhancements/8515-support-custom-request-to-limit-ratio/README.md

```diff
@@ -0,0 +1,216 @@
+   * **Recreate mode**: When a new request-to-limit ratio is set, the ratio is applied only on Pod creation, after the Updater evicts the running Pod. In this mode, updating the request-to-limit ratio on a running Pod will affect the limits only after the Pod is evicted (either by the Updater or manually, e.g. via `kubectl delete pod`) when the current `resources.requests` differ significantly from the new recommendation.  
+   * **InPlaceOrRecreate mode** (alpha in v1.4.0): When a new request-to-limit ratio is set, the VPA Updater will attempt in-place updates using the `/resize` subresource to modify `Pod.Spec.Containers[i].Resources.limits` or `Pod.Spec.Containers[i].Resources.requests` in certain situations. If the in-place update fails, it falls back to evicting the Pod and performing a recreation. For more details, see the [In-Place Updates documentation](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/docs/features.md#in-place-updates-inplaceorrecreate).  
+   * **Initial mode**: VPA updates the request-to-limit ratio only during Pod creation and does not change it later.
+2. If the `RequestToLimitRatio` feature gate is disabled, the request-to-limit ratio already set in the workload API (e.g. Deployment API) is used.
+
+> [!IMPORTANT]  
+> This new feature can be used together with other features in development, such as the [fixed memory-per-CPU ratio feature](https://github.com/kubernetes/autoscaler/pull/8459).
+
+### Validation
+
+  * To use this functionality, the `RequestToLimitRatio` feature flag must be enabled.
+
+#### Static Validation via CRD Rules
+
+* The `RequestToLimitRatio` configuration will be validated when VPA CRD objects are created or updated. For example:  
+  * If `Type` is `Factor`, the value must be greater than or equal to 1 (enforced via CRD validation rules).  
+  * If `Type` is `QuantityPercentage`, the value must be greater than or equal to 1 (enforced via CRD validation rules).  
+
+#### Dynamic Validation via Admission Controller
+
+* When using the new `RequestToLimitRatio` field, the `controlledValues` field must be set to `RequestsAndLimits`. It does not make sense to specify `RequestToLimitRatio` if VPA is not allowed to update limits. This requirement is enforced by the admission controller.  
+* If `Type` is set to `QuantityValue`, then its `Value` will be validated.
+
+### Feature Enablement and Rollback
+
+#### Enabling or Disabling the Feature in a Live Cluster
+
+* Enable the feature by setting the `RequestToLimitRatio` feature gate.  
+* Components affected by this feature gate:  
+  * admission-controller  
+  * updater
+
+#### When Enabled
+
+* The admission controller will **accept** new VPA objects that include a configured `RequestToLimitRatio`.  
+* For containers targeted by a VPA object using `RequestToLimitRatio`, the admission controller and/or the updater will enforce the configured ratio.
```

Comment:
What happens in a scenario like this:
1. A user has a pod with in-place set. And where requests are equal to limits (and the QoS class is Guaranteed)
2. The user modifies the VPA to add a RequestToLimitRatio, making limits double requests

What should the behaviour be? Should the VPA kill the pod to allow a new one be created with a different QoS class?

Reply:
I guess my concern is actually more of "how will the VPA handle the RequestToLimitRatio changing". Either from the default to a specified ratio, or from one ratio to another.
I think it's worth calling this out in the AEP